packages/grid/src/GridModel.ts

```diff
@@ -135,7 +158,36 @@ abstract class GridModel<
    * @param column Column to check
    * @returns True if the column is movable
    */
-  isColumnMovable(column: ModelIndex): boolean {
+  isColumnMovable(column: ModelIndex, depth = 0): boolean {
+    return true;
+  }
+
+  /**
+   * Checks if a column can be moved to a specific index
+   * @param fromIndex The index to move from
+   * @param toIndex The index to move to
+   * @param depth The depth the movement is occurring at
+   * @returns If the column can be moved to the given index at the given depth
+   */
+  isColumnMovableTo(column: ModelIndex, to: ModelIndex, depth = 0): boolean {
+    return true;
+  }
+
+  /**
+   * Checks if a column can be dropped between 2 columns
+   * Assumes the left and right indexes are adjacent to each other
+   * @param fromIndex The index to move from
+   * @param leftIndex The index on the left of the drop spot
+   * @param rightIndex The index on the right of the drop spot
+   * @param depth The depth the movement is occurring at
+   * @returns If the column can be dropped between the given indexes at the given depth
+   */
+  isColumnDroppableBetween(
```

Comment:
`isColumnMovableTo` is whether the column is within the bounds of its parent basically.

`isColumnDroppableBetween` is if a destination for dropping is valid. This is needed if it's at the model level because the model only knows about model indexes. So when dragging in an instance like below, how do you tell where `Char5` can be dropped?

![image](https://user-images.githubusercontent.com/8619789/178664632-d09ee4cd-aa84-45b7-bf49-6da81871ee9e.png)

It can be dropped at visible index 1 here, even though visible index 1 is not in the same parent group as it. And it can be dropped at 6 even thought it is also not in the same group. But not at 2, 3, or 5. Now do that without knowledge of visible indexes and the only way is to know what model indexes you're dropping between.

I guess the logic of both could be moved to the column drag handler now since I added a `getVisibleRange` to the column group class. Do you think it makes more sense in the drag handler then?

Reply:
Moved handling of this to drag handler